ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Round.java

```diff
@@ -0,0 +1,139 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.math;
+
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.function.udf.UdfSchemaProvider;
+import io.confluent.ksql.schema.ksql.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.KsqlException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.List;
+
+/*
+The rounding behaviour implemented here follows that of java.lang.Math.round()
+It's an implementation of rounding "half up". This means we round to the nearest integer value and
+in the case we are equidistant from the two nearest integers we round UP to the nearest integer.
+This means:
+
+ROUND(1.1) -> 1
+ROUND(1.5) -> 2
+ROUND(1.9) -> 2
+
+ROUND(-1.1) -> -1
+ROUND(-1.5) -> -1    Note this is not -2! We round up and up is in a positive direction.
+ROUND(-1.9) -> -2
+ */
+@UdfDescription(name = "Round", description = Round.DESCRIPTION)
+public class Round {
+
+  static final String DESCRIPTION =
+      "Returns the value rounded to the specified number of decimal places. "
+      + "If the number of decimal places is omitted it defaults to zero."
+      + "Numbers are rounded to the nearest digit at the last required decimal place. "
+      + "If the digit to the right of the last required decimal place is equidistant (i.e. a 5) "
+      + "then it is rounded up.";
+
+  private static final MathContext SIMPLE_ROUND_HALF_UP =
+      new MathContext(1, RoundingMode.HALF_UP);
+
+  @Udf
+  public Long round(@UdfParameter final Double val) {
+    return val == null ? null : Math.round(val);
+  }
+
+  @Udf
+  public Double round(@UdfParameter final Double val, @UdfParameter final Integer decimalPlaces) {
+    // Note that we MUST BigDecimal.valueOf(double) to create the BD to ensure correctness
+    // as this methods initialises via the string representation
+    return val == null
+        ?
+        null : roundBigDecimal(BigDecimal.valueOf(val), decimalPlaces).doubleValue();
+  }
+
+  @Udf(schemaProvider = "provideDecimalSchema")
+  public BigDecimal round(@UdfParameter final BigDecimal val) {
+    return val == null ? null : round(val, 0);
+  }
+
+  @Udf(schemaProvider = "provideDecimalSchemaWithDecimalPlaces")
+  public BigDecimal round(
+      @UdfParameter final BigDecimal val,
+      @UdfParameter
+      final Integer decimalPlaces
+  ) {
+    return val == null ? null : roundBigDecimal(val, decimalPlaces);
+  }
+
+  @UdfSchemaProvider
+  public SqlType provideDecimalSchemaWithDecimalPlaces(final List<SqlType> params) {
+    if (params.size() != 2) {
+      throw new KsqlException("Round udf with decimal places accepts two parameters");
```

Comment:
This would be a programming error. Not a sure error. So don't throw `KsqlException`, (which is for user errors). Throw `IllegalArgumentException`.  (Can you also fix `Abs` so no one else copies it, please?).

Same below.

Though, tbh, given you're not actually doing anything with the params... I wonder what the point of the error checking is?  If ever UDF did the same thing that would just be a whole bunch of production code that was doing nothing more than testing it was called with the right params. We have unit tests that test the same... so, consider removing these checks.

Reply:
This is just copy and paste from abs ;)

Reply:
done